mmcv/ops/merge_cells.py

```diff
@@ -42,18 +43,18 @@ class BaseMergeCell(nn.Module):
     """
 
     def __init__(self,
-                 fused_channels=256,
-                 out_channels=256,
-                 with_out_conv=True,
-                 out_conv_cfg=dict(
+                 fused_channels: Optional[int] = 256,
+                 out_channels: Optional[int] = 256,
```

Comment:
`Optional` is only used when the default value is None,

Reply:
>  if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not.

refers to https://docs.python.org/3/library/typing.html#typing.Optional

Reply:
> Optional[X] is equivalent to X | None (or Union[X, None]).
refers to https://docs.python.org/3/library/typing.html#typing.Optional
Your understanding may be slightly biased.

Reply:
fused_channels or out_channels can be None when with_out_conv is False so Optional[int] is correct.

I create a [pr](https://github.com/open-mmlab/mmcv/pull/2062) to ensure fused_channels or out_channels can not be None when with_out_conv is True.